### ISSUE TITLE
IDE0301/IDE0303/IDE0305 triggers on code that cannot be converted

### ISSUE THREAD

**Issue by bradwilson** [dotnet/roslyn#75885]
**Version Used**:

.NET 9 SDK 9.0.100 (note: does not trigger with .NET 8 SDK)

**Steps to Reproduce**:

Have two projects share source code; one targets `Microsoft.CodeAnalysis.CSharp` version `3.11.0`, the other targets `4.11.0`.

_For IDE0301: ([example](https://github.com/xunit/xunit.analyzers/blob/5c60a820b3ebed5a26b02bfc2b69780c9ae6c065/src/xunit.analyzers/Utility/SerializableTypeSymbols.cs#L50-L52))_
 
1. Write a function that returns `ImmutableArray<INamedTypeSymbol>`
2. Write this line of code in the function:
   ```csharp
   return ImmutableArray<INamedTypeSymbol>.Empty;
   ```

_For IDE0303: ([example](https://github.com/xunit/xunit.analyzers/blob/5c60a820b3ebed5a26b02bfc2b69780c9ae6c065/src/xunit.analyzers/X1000/InlineDataMustMatchTheoryParameters.cs#L79-L81))_

1. Use `ImmutableArray.Create(singleValue)`

_For IDE0305: ([example](https://github.com/xunit/xunit.analyzers/blob/5c60a820b3ebed5a26b02bfc2b69780c9ae6c065/src/xunit.analyzers/Utility/XunitDiagnosticAnalyzer.cs#L14-L16))_

1. Convert a standard array to immutable with `someArray.ToImmutableArray()`

**Diagnostic Id**:

`IDE0301: Collection initialization can be simplified`
`IDE0303: Collection initialization can be simplified`
`IDE0305: Collection initialization can be simplified`

**Expected Behavior**:

Do not trigger on this code.

**Actual Behavior**:

Analyzer incorrectly triggers. The fixes will not compile in the Roslyn 3.11 linked code, as the compiler fails with `CS9210: This version of 'ImmutableArray<T>' cannot be used with collection expressions.`.

**Comment by CyrusNajmabadi** [comment]
This is by design. Analyzers run per compilation, independently. So the diagnostics and fixes are correct for the later version. 

If you are multi targeting, and you don't want this situation, you can disable these analyzers from running.
